tests/helpers/TestClusterChecker.py

```diff
@@ -43,109 +43,58 @@ def test_checkExistingClusters_empty(self):
         self.checker.checkExistingClusters()
         expected = [call.listMongoObjects()]
         self.assertEqual(expected, self.kubernetes_service.mock_calls)
-        self.assertEqual({}, self.checker.cluster_versions)
+        self.assertEqual({}, self.checker._cluster_versions)
 
     def test_checkExistingClusters_bad_format(self):
         self.kubernetes_service.listMongoObjects.return_value = {"items": [{"invalid": "object"}]}
         self.checker.checkExistingClusters()
         expected = [call.listMongoObjects()]
         self.assertEqual(expected, self.kubernetes_service.mock_calls)
-        self.assertEqual({}, self.checker.cluster_versions)
+        self.assertEqual({}, self.checker._cluster_versions)
 
-    @patch("mongoOperator.helpers.BackupChecker.BackupChecker.backupIfNeeded")
-    def test_checkExistingClusters(self, backup_mock):
-        self.checker.cluster_versions[("mongo-cluster", "default")] = "100"  # checkCluster will assume cached version
+    @patch("mongoOperator.services.MongoService.MongoClient")
+    @patch("mongoOperator.helpers.BackupHelper.BackupHelper.backupIfNeeded")
+    def test_checkExistingClusters(self, backup_mock, mongo_client_mock):
+        self.checker._cluster_versions[("mongo-cluster", self.cluster_object.metadata.namespace)] = "100"
         self.kubernetes_service.listMongoObjects.return_value = {"items": [self.cluster_dict]}
-        self.kubernetes_service.execInPod.return_value = self._getMongoFixture("replica-status-ok")
+        mongo_client_mock.return_value.admin.command.return_value = self._getMongoFixture("replica-status-ok")
         self.checker.checkExistingClusters()
-        self.assertEqual({("mongo-cluster", "default"): "100"}, self.checker.cluster_versions)
-        expected = [call.listMongoObjects(),
-                    call.execInPod('mongodb', 'mongo-cluster-0', 'default',
-                                   ['mongo', 'localhost:27017/admin', '--eval', 'rs.status()'])]
+        self.assertEqual({("mongo-cluster", self.cluster_object.metadata.namespace): "100"},
+                         self.checker._cluster_versions)
+        expected = [call.listMongoObjects()]
         self.assertEqual(expected, self.kubernetes_service.mock_calls)
         backup_mock.assert_called_once_with(self.cluster_object)
 
-    @patch("mongoOperator.helpers.ClusterChecker.ClusterChecker.checkCluster")
-    @patch("kubernetes.watch.watch.Watch.stream")
-    def test_streamEvents_add_update(self, stream_mock, check_mock):
```

Comment:
Don't we support these events anymore?

Reply:
Kubernetes event stream was never used (and now removed) because it seemed unstable when doing tests with it.